TypeCobol.Analysis/Cfg/ControlFlowGraphBuilder.PerformCallRelation.cs

```diff
@@ -0,0 +1,202 @@
+            }
+            /// <summary>
+            /// Add a PERFORM CALL relation : Caller => Callee (Procedure)
+            /// Example of code:
+            ///        main.
+            ///             perform a
+            ///             goback
+            ///             .
+            ///        a.
+            ///             perform b
+            ///        .
+            /// The relation is: a => b
+            /// </summary>
+            /// <param name="performCaller">PERFORM procedure caller, source instruction: in the example (perform a)</param>
+            /// <param name="caller">Caller procedure: in the example 'a'</param>/// 
+            /// <param name="performCallee">The PERFORM procedure called, target instruction: in the example (perform b)</param>
+            /// <param name="callee">The called procedure: in the example 'b' </param>
+            internal void AddToRelation(PerformProcedure performCaller, Procedure caller, PerformProcedure performCallee, Procedure callee)
+            {
+                if (!TryGetValue(caller, out var performCallChain))
+                {
+                    performCallChain = new PerformCallChain(performCaller);
+                    Add(caller, performCallChain);
+                }
+                performCallChain.Procedures.Add(callee);
+                if (!performCallChain.PerformCallees.Contains(performCallee))
+                {
+                    performCallChain.PerformCallees.Add(performCallee);
+                }
+                Domain.Add(caller);
+                Domain.Add(callee);
+            }
+
+            /// <summary>
+            /// Compute the transitive closure of the PERFORM CALL relation.
+            /// We browse the the graph represented using the Call Relation, then we found during the visit cycles.
+            /// In the same time we compute the call chain.
+            /// </summary>
+            internal void TransitiveClosure()
+            {
+                // The dfs stack of all procedures in active consideration.
+                Stack<Procedure> S = new Stack<Procedure>();
+                // The dictionary N is used for three purposes
+                // 1) To record unmarked procedures (N[p] == 0).
+                // 2) To associate a positive number value with procedures that are active consideration (0 < N[p] < Infnity).
+                // 3) To mark Infinity Procedures whose cycles have already been found
+                Dictionary<Procedure, int> N = new Dictionary<Procedure, int>();
+
+                // For each procedure in the relation's domain
+                foreach (var a in Domain)
+                { // For each procedure 'a' not yet considered, treat it.
+                    var na = Visited(a);
+                    if (na == 0)
+                        dfs(a);
+                }
+
+                // Depth First Search Traversal.
+                void dfs(Procedure a)
```

Comment:
This last version of the code without Tuple is much clearer.
Can you use more meaningful name for variables?
Like `procedure` or at least `proc` instead of `a` for the name of the procedure.
Same remarks for variables : `d`, `fa`, `b`, `nb`, ...

Reply:
Done in 35d0734